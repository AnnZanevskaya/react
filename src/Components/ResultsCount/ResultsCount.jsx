import React from "react";
import PropTypes from 'prop-types';

const ResultsCount = (props) => (
    <h4 className="results__description"><span className="results__count">{props.count}</span> movies found</h4>
);

ResultsCount.propTypes = {
    count: PropTypes.number.isRequired,
}

ResultsCount.defaultProps = {
    count: 0
}

export default ResultsCount;
