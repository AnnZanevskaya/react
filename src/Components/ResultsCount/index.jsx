import React from "react";

const ResultsCount = () => (
    <h4 className="results__description"><span className="results__count">39</span> movies found</h4>
)

import default ResultsCount;