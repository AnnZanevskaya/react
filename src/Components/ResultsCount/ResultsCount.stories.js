import React from 'react';
import { withKnobs,number } from '@storybook/addon-knobs';

import ResultsCount from './ResultsCount';
import '../../Styles/sass/style.scss';

export default {
  title: 'ResultsCount',
  component: ResultsCount,
  decorator: [withKnobs]
};

const Template = (args) => <ResultsCount {...args} />;

export const ResultsCountElement = Template.bind({});

ResultsCount.args = {
    count: number("Count", 5)
};
