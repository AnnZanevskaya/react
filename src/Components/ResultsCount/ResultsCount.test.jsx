import React from 'react';
import renderer from 'react-test-renderer';
import ResultsCount from './ResultsCount';

it('renders correctly', () => {
  const tree = renderer
    .create(<ResultsCount count={2} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});