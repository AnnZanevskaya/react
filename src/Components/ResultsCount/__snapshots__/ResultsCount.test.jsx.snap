// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<h4
  className="results__description"
>
  <span
    className="results__count"
  >
    2
  </span>
   movies found
</h4>
`;
